### PR TITLE
Public the StateCircuit fields 

### PR DESCRIPTION
**Pull request by Lawliet-Chan**
To build `StateCircuit` for external project. Same reason as https://github.com/appliedzkp/zkevm-circuits/pull/262

### CODE REVIEW

**Review by CPerezz — COMMENTED** [review]
I'm maybe more biased to impl a `new` or `from_ops` constructor instead of just expose the fields. 

As they're not intended to be modified once the Circuit instance has been build with the ops.

@han0110  any suggestions?

**Comment by Lawliet-Chan** [comment]
> 

I see, it sounds better.

File: zkevm-circuits/src/state_circuit/state.rs [hunk 1]
```diff
@@ -1167,6 +1167,37 @@ pub struct StateCircuit<
...
+        GLOBAL_COUNTER_MAX,
+        MEMORY_ROWS_MAX,
+        MEMORY_ADDRESS_MAX,
+        STACK_ROWS_MAX,
+        STACK_ADDRESS_MAX,
+        STORAGE_ROWS_MAX,
+    >
+{
+    #[allow(missing_docs)]
+    pub fn new(
```
Comment (CPerezz) [inline zkevm-circuits/src/state_circuit/state.rs:1188]: Could you just add some doc lines mentioning that the fn generates a new `StateCircuit` instance containing the operations passed?

On that way we don't need the `#[allow(...)]`

**Review by CPerezz — CHANGES_REQUESTED** [review]

File: zkevm-circuits/src/state_circuit/state.rs (hunk 1 above)
Comment (Lawliet-Chan) [reply zkevm-circuits/src/state_circuit/state.rs:1188]: SURE

**Review by CPerezz — APPROVED** [review]
LGTM!!! 

Thanks @Lawliet-Chan  :smile:

**Pull request merged**